### ISSUE TITLE
Add support for multi attachment

### ISSUE THREAD

**Issue by FikriMilano** [google/android-fhir#1797]
**Is your feature request related to a problem? Please describe.**
The current Attachment widget is only able to handle single attachment.

**Describe the solution you'd like**
Update Attachment widget to be able to handle multi attachment.

**Describe alternatives you've considered**
N/A

**Additional context**
<img src="https://user-images.githubusercontent.com/86378921/210457526-44ad7c16-347f-4bd9-8f2b-9f1c42d2b060.png" width="400">

**Would you like to work on the issue?**
Yes

**Comment by FikriMilano** [comment]
@f-odhiambo @jingtang10

**Comment by FikriMilano** [comment]
This ticket is not ready for development yet

**Comment by FikriMilano** [comment]
My main concern is when submitting Questionnaire, the answers will get too big because of the attachments, which will cause the app to crash.

**Comment by FikriMilano** [comment]
Perhaps we need some changes when submitting a Questionnaire, if it's too big, then process it in small parts

**Comment by FikriMilano** [comment]
In the past we have encountered this problem in FHIRCore as well, since it's not SDK, there's no way to change the process when submitting a Questionnaire.

But here in SDK, I hope we can address this first, before going forward with this ticket

**Comment by FikriMilano** [comment]
Summary from earlier discussion, we can submit the base64 encoded file into one of the mentioned FHIR resource below, and append the link into the [Attachment Url](https://build.fhir.org/datatypes-definitions.html#Attachment.url).
- [Binary](https://hl7.org/fhir/binary.html) resource
Heavy load on the database, because files may be large.
May not be feasible for super large files e.g. video.

- [DocumentReference](https://hl7.org/fhir/documentreference.html) resource
Can be located outside of database, which reduce the heavy load on the database.
May need another endpoint to sync files into server.


@jingtang10 @f-odhiambo @dubdabasoduba @aditya-07

**Comment by dubdabasoduba** [comment]
@jingtang10 does it makes sense to also have max file size for the SDK? I don't think we want to allow people to upload GBs of data.

**Comment by FikriMilano** [comment]
@dubdabasoduba yeah, we have the maxSize merged https://github.com/google/android-fhir/pull/1688

**Comment by FikriMilano** [comment]
Blocked by https://github.com/google/android-fhir/issues/1858

**Comment by jingtang10** [comment]
hey @FikriMilano what needs to be done in this ticket
